components/catalogs/back/src/Infrastructure/Persistence/FindOneAttributeByCodeQuery.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Infrastructure\Persistence;
+
+use Akeneo\Pim\Structure\Component\Model\AttributeInterface;
+use Akeneo\Pim\Structure\Component\Repository\AttributeRepositoryInterface;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class FindOneAttributeByCodeQuery
+{
+    public function __construct(private AttributeRepositoryInterface $repository)
+    {
+    }
+
+    /**
+     * @return array{code: string, label: string, type: string, scopable: bool, localizable: bool}
+     */
+    public function execute(string $code): ?array
+    {
+        /** @var AttributeInterface|null $attribute */
+        $attribute = $this->repository->findOneByIdentifier($code);
+
+        if (!$attribute) {
```

Comment:
```suggestion
        if (null === $attribute) {
```
I rather want to be explicit